ext/IntegralsGSLExt.jl

```diff
@@ -0,0 +1,48 @@
+module IntegralsGSLExt
+
+using GSL
+using Integrals
+using Integrals: IntegralCache
+
+mutable struct GSLCache{T}
+    value::T
+end
+getvalue(cache::GSLCache) = cache.value
+
+function Integrals.init_cacheval(alg::GSLIntegration{typeof(integration_cquad)}, prob::IntegralProblem)
+    ws = integration_cquad_workspace_alloc(alg.kws.wssize)
+    gslcache = GSLCache(ws)
+    finalizer(integration_cquad_workspace_free∘getvalue, gslcache)
```

Comment:
Assuming this file is fine https://github.com/JuliaMath/GSL.jl/blob/fb44df54f46622dd923850566d422e1a8f46691b/examples/Quadrature.jl#L9 this is the correct way to use the finalizer yes